api/v1/api.go

```diff
@@ -210,3 +209,71 @@ func (api *API) respondError(w http.ResponseWriter, apiErr apiError, data interf
 		level.Error(api.logger).Log("msg", "failed to write data to connection", "err", err)
 	}
 }
+
+type metricJson map[string]interface{}
+
+func metricsToInterface(metrics []*dto.Metric, metricsType dto.MetricType) []metricJson {
```

Comment:
`metricsToInterface` is a bit too generic. Perhaps `makeEncodable` or `makeEncodableMetrics`?